src/Components/ServiceScene/JSONPanel/ValueRenderer.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import { AdHocFilterWithLabels } from '@grafana/scenes';
+
+import { logsSyntaxMatches } from '../../../services/logsSyntaxMatches';
+import { JsonDataFrameLabelsName, JsonDataFrameStructuredMetadataName, JsonDataFrameTimeName } from '../LogsJsonScene';
+import { highlightLineFilterMatches, highlightRegexMatches } from './highlightLineFilterMatches';
+import { KeyPath } from '@gtk-grafana/react-json-tree/dist/types';
+
+interface ValueRendererProps {
+  keyPath: KeyPath;
+  lineFilters: AdHocFilterWithLabels[];
+  valueAsString: unknown;
+}
+
+export default function ValueRenderer({ keyPath, lineFilters, valueAsString }: ValueRendererProps) {
+  if (keyPath[0] === JsonDataFrameTimeName) {
+    return null;
+  }
+  const value = valueAsString?.toString();
+  if (!value) {
+    return null;
+  }
+
+  const keyPathParent = keyPath[1];
+
+  if (
+    keyPathParent !== undefined &&
+    keyPathParent !== JsonDataFrameStructuredMetadataName &&
+    keyPathParent !== JsonDataFrameLabelsName
+  ) {
+    let valueArray = highlightLineFilterMatches(lineFilters, value);
+
+    // If we have highlight matches we won't show syntax highlighting
+    if (valueArray.length) {
+      return valueArray.filter((e) => e);
```

Comment:
Is this actually a:
```suggestion
      return valueArray.slice();
```
?

Reply:
I'm not sure I understand, but I don't think this is necessary any more, I think this was a carry over from a previous implementation that could contain null or undefined values in the array we didn't want to render. Fixed in https://github.com/grafana/logs-drilldown/pull/1382/commits/c3d81cd52e20ec05400a25e6dd1cc1eb4c275ed9